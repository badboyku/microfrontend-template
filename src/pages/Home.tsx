import * as React from 'react';
import HelloWorld from '../components/HelloWorld';

type Props = {};

const Home: React.FC<Props> = (_props) => {
  return (
    <div>
      <HelloWorld />
    </div>
  );
};

export default Home;
