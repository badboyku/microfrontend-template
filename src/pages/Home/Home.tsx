import * as React from 'react';
import HelloWorld from '../../components/HelloWorld';

type Props = {};

const Home = (_props: Props): JSX.Element => {
  return (
    <div>
      <HelloWorld />
    </div>
  );
};

export default Home;
