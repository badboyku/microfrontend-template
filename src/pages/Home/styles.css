.logo {
  width: 40px;
}

.envVarsContainer {
  display: flex;
  flex-direction: column;
  gap: 2px;
}
.envVar {
  display: flex;
  flex-direction: row;
  gap: 10px;
}
.envVarName {
  width: 225px;
  min-width: 225px;
  max-width: 225px;
  white-space: nowrap;
}
.envVarValue {
  font-weight: bold;
  word-break: break-all;
}
