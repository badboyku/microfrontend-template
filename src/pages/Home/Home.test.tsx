import { render, screen } from '@testing-library/react';
import Home from './Home';

jest.mock('components/HelloWorld', () => () => <div data-testid="helloWorld" />);
jest.mock('utils/logger');

const renderHome = (props = {}) => {
  return render(<Home {...props} />);
};

describe('Page Home', () => {
  it('renders without crashing', () => {
    const { asFragment } = renderHome();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders header Home', () => {
    renderHome();

    expect(screen.getByText('Home')).toBeInTheDocument();
  });

  it('renders component HelloWorld', () => {
    renderHome();

    expect(screen.getByTestId('helloWorld')).toBeInTheDocument();
  });
});
