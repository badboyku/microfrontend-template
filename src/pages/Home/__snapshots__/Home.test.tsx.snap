// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Page Home renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Home
    </h1>
    <div
      data-testid="helloWorld"
    />
    <img
      alt="logo"
      class="logo"
      src="logo.svg"
    />
    <svg
      width="40"
    >
      logo.svg
    </svg>
    <h3>
      Build Time Env Vars:
    </h3>
    <div
      class="envVarsContainer"
    >
      <div
        class="envVar"
      >
        <div
          class="envVarName"
        >
          IS_DEV:
        </div>
        <div
          class="envVarValue"
        >
          false
        </div>
      </div>
      <div
        style="display: flex; flex-direction: row; gap: 10px;"
      >
        <div
          class="envVarName"
        >
          IS_PROD:
        </div>
        <div
          class="envVarValue"
        >
          true
        </div>
      </div>
    </div>
    <h3>
      Runtime Env Vars:
    </h3>
    <div
      class="envVarsContainer"
    >
      <div
        class="envVar"
      >
        <div
          class="envVarName"
        >
          NODE_ENV:
        </div>
        <div
          class="envVarValue"
        >
          test
        </div>
      </div>
      <div
        class="envVar"
      >
        <div
          class="envVarName"
        >
          REACT_APP_MY_VAR:
        </div>
        <div
          class="envVarValue"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
